# Description: workaround for the lts, we don't want to expose 
#  gnome-online-accounts to users in Unity but we still want to be able to
#  open it from softwares which have an account configured
# Upstream: the change is ubuntu specific
Index: gnome-control-center-3.3.91/shell/gnome-control-center.c
===================================================================
--- gnome-control-center-3.3.91.orig/shell/gnome-control-center.c	2012-03-06 17:52:14.883083003 +0100
+++ gnome-control-center-3.3.91/shell/gnome-control-center.c	2012-03-06 17:52:20.695083286 +0100
@@ -926,6 +926,21 @@
                                              &iter);
     }
 
+  if (!name && g_strcmp0(start_id, "online-accounts") == 0)
+    {
+        char *desktop = "/usr/share/applications/gnome-online-accounts-panel.desktop";
+        if (!g_file_test (desktop, G_FILE_TEST_EXISTS)) 
+          {
+            g_warning ("Could not find settings panel \"%s\"", start_id);
+            return FALSE;
+          }
+        GAppInfo*appinfo = g_desktop_app_info_new_from_filename (desktop);
+        const gchar *name = g_app_info_get_name (appinfo);
+        GIcon       *icon = g_app_info_get_icon (appinfo);
+        activate_panel (GNOME_CONTROL_CENTER (shell), start_id, argv, desktop, name, icon);
+        g_object_unref(appinfo);
+    }
+
   if (!name)
     {
       g_warning ("Could not find settings panel \"%s\"", start_id);
Index: gnome-control-center-3.3.91/panels/online-accounts/gnome-online-accounts-panel.desktop.in.in
===================================================================
--- gnome-control-center-3.3.91.orig/panels/online-accounts/gnome-online-accounts-panel.desktop.in.in	2012-03-05 15:04:55.000000000 +0100
+++ gnome-control-center-3.3.91/panels/online-accounts/gnome-online-accounts-panel.desktop.in.in	2012-03-06 17:52:14.907083003 +0100
@@ -7,7 +7,7 @@
 Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;
-OnlyShowIn=GNOME;Unity;
+OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
 X-GNOME-Bugzilla-Component=Online Accounts
